scripts/controllers/usefulness_controller.ts

```diff
@@ -0,0 +1,58 @@
+import { Controller } from '@hotwired/stimulus'
+
+class UsefulnessController extends Controller {
+    vote: String | null
+    launchModal: String | null
+    
+    initialize() {
+      this.vote = null
+      this.launchModal = ''
+    }
+  
+    static get targets() {
+      return []
+    }
+  
+    handleVote(event: any) {
+      this.launchModal = event.params.launchModal
+      this.vote = this.launchModal === 'useful-yes' ? 'yes' : 'no'
+      const modal = document.querySelector(`#${this.launchModal}`)
+      modal.classList.add('active')
+    }
+  
+    handleCancel() {
+      this.closeModal()
+    }
+  
+    handleSubmit() {
+      const documentId = this.data.get('documentId')
+      const location = this.vote === 'yes' ? '/api/upvote' : '/api/downvote'
```

Comment:
kinda seems like you only store `this.vote` based on `this.launchModal` for just this one spot. why not have this check the `this.launchModal` value?